### PR TITLE
ENH: Added NX_class as a PerObjectHint to the event descriptor.

### PR DESCRIPTION
**Pull request by burkeds**
This is an optional hint that can be used to carry a NeXus class definition one may wish to associate with a device.

## Motivation and Context
It would be good to have a standard way to assign a NeXus class definition to a high level device. Having a specific field with a standard name will better allow the community to converge on a standard way to assign and use this definition for high level processes such as writing callbacks.

### CODE REVIEW

**Review by danielballan — APPROVED** [review]
This looks good to me, but @evalott100 should sign off before we merge, to confirm that there's nothing special we have to do with schema generation here.

**Comment by evalott100** [comment]
> there's nothing special we have to do with schema generation here

CI does our generation check! LGTM :pray:

**Review by evalott100 — APPROVED** [review]

**Comment by danielballan** [comment]
Nice to have a @burkeds commit in event-model. Thanks Devin!

**Pull request merged**
